jax/_src/numpy/lax_numpy.py

```diff
@@ -1958,6 +1966,14 @@ def block(arrays: ArrayLike | list[ArrayLike]) -> Array:
   out, _ = _block(arrays)
   return out
 
+@overload
```

Comment:
Provide overloads to narrow the return type.

No overload is provided for `atleast_1d()`, but we could provide it if desired?

Reply:
These `atleast_*()` overloads seem like a useful change that could be done in their own PR.

Since this is a public API, we'll have to do a round of tests and fix the downstream failures that it causes (making annotations more strict, even when it's correct, in general uncovers incorrect annotations in downstream packages).